tools/dep_updaters/update-zlib.sh

```diff
@@ -1,23 +1,47 @@
 #!/bin/sh
 set -e
-# Shell script to update zlib in the source tree to a specific version
+# Shell script to update zlib in the source tree to the most recent version.
+# Zlib rarely creates tags or releases, so we use the latest commit on the main branch.
+# See: https://github.com/nodejs/node/pull/47417
 
 BASE_DIR=$(cd "$(dirname "$0")/../.." && pwd)
 DEPS_DIR="$BASE_DIR/deps"
 
-CURRENT_VERSION=$(grep "#define ZLIB_VERSION" "$DEPS_DIR/zlib/zlib.h" | sed -n "s/^.*VERSION \"\(.*\)\"/\1/p")
+echo "Comparing latest upstream with current revision"
 
-NEW_VERSION_ZLIB_H=$(curl -s "https://chromium.googlesource.com/chromium/src/+/refs/heads/main/third_party/zlib/zlib.h?format=TEXT" | base64 --decode)
+# Revert zconf.h changes before checking diff 
+perl -i -pe 's|^//#include "chromeconf.h"|#include "chromeconf.h"|' "$DEPS_DIR/zlib/zconf.h"
+git stash -- "$DEPS_DIR/zlib/zconf.h"
 
-NEW_VERSION=$(printf '%s' "$NEW_VERSION_ZLIB_H" | grep "#define ZLIB_VERSION" | sed -n "s/^.*VERSION \"\(.*\)\"/\1/p")
+git fetch https://chromium.googlesource.com/chromium/src/third_party/zlib.git HEAD
```

Comment:
Since the CQ seems to think the CI needs to be re-run anyway (it says `✘  Last GitHub CI failed`, which I think is a bug with GH), I suggest we push another commit to put the `git fetch` before the `git stash`, in case e.g. the machine doesn't have internet, we don't "pollute" the stash with an entry that's never cleaned up.

Reply:
Lately CQ has been acting weird requesting CI where its not usually needed or thinking it failed. Should I open an issue about it?

Reply:
> I suggest we push another commit to put the git fetch before the git stash, in case e.g. the machine doesn't have internet, > we don't "pollute" the stash with an entry that's never cleaned up.

@aduh95 [done here](https://github.com/nodejs/node/pull/48054/commits/c7650e681df025fab7c4f8169c9b42ae73a838ef)